audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -168,79 +169,81 @@ static MediaMetadataCompat getMediaMetadata(String mediaId) {
         return mediaMetadataCache.get(mediaId);
     }
 
-    Bitmap loadArtBitmap(String artUriString, String loadThumbnailUri) {
-        Bitmap bitmap = artBitmapCache.get(artUriString);
-        if (bitmap != null) return bitmap;
-        try {
-            // There are 3 cases handled by this function:
-            //   1. content URI with openFileDescriptor
-            //   2. content URI with loadThumbnail (when Android >= Q and specified by the config)
-            //   3. not content URI - loading from the file, or cache file created by the Dart side
-            Uri artUri = Uri.parse(artUriString);
-            boolean usesContentScheme = "content".equals(artUri.getScheme());
-            FileDescriptor fileDescriptor = null;
-            if (usesContentScheme) {
-                try {
-                    if (loadThumbnailUri != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                        Size defaultSize = new Size(192, 192);
-                        bitmap = getContentResolver().loadThumbnail(
-                                artUri,
-                                new Size(config.artDownscaleWidth == -1
-                                                ? defaultSize.getWidth()
-                                                : config.artDownscaleWidth,
-                                        config.artDownscaleHeight == -1
-                                                ? defaultSize.getHeight()
-                                                : config.artDownscaleHeight),
-                                null);
-                        if (bitmap == null) {
-                            return null;
-                        }
+    // This will rethrow any errors it will encounter during loading,
+    // so that Dart side can detect that and load default art instead.
+    Bitmap loadArtBitmap(String artUriString, String loadThumbnailUri, boolean updateFallbackArtCache) {
+        Bitmap bitmap = null;
+        if (!updateFallbackArtCache) {
+            Bitmap cachedBitmap = artBitmapCache.get(artUriString);
+            if (cachedBitmap != null) return cachedBitmap;
+        }
```

Comment:
New parameter needed to forcefully update the art when `setFallbackArt` is called